File: server/ruby/app/infrastructure/mappers/payment_details/api_error_mapper.rb
module Infrastructure
  module Mappers
    module PaymentDetails
      class ApiErrorMapper
        def self.from_sdk_response(response)
          return nil if response.nil?

          dto = Business::Domain::Payments::PaymentDetails::APIError.new
          dto.message = response.message
          dto.error_code = response.error_code
          dto.property_name = response.property_name
          dto.http_status_code = response.http_status_code
          dto.retriable = response.retriable
          dto.category = response.category
          dto.id = response.id

          dto
        end
      end
    end
  end
end

File: server/nodejs/src/infrastructure/mappers/payment-details/api-error-mapper.ts
import type { Domain } from 'onlinepayments-sdk-nodejs';
import { APIError } from '../../../business/domain/payments/payment-details/api-error';

type APIErrorSdk = Domain.APIError;

export const ApiErrorMapper = {
    fromSdkResponse: (response?: APIErrorSdk): APIError | undefined => {
        if (!response) {
            return undefined;
        }

        return {
            message: response?.message,
            errorCode: response?.errorCode,
            propertyName: response?.propertyName,
            httpStatusCode: response?.httpStatusCode,
            retriable: response?.retriable,
            category: response?.category,
            id: response?.id,
        };
    },
};
